### PR TITLE
Miscellaneous fixes...

### PR DESCRIPTION
**Pull request by nphilipp**
...what I said on standup that day wasn't just empty promises. It just keeps on growing. :wink:

* Vagrant: fix how the IPA server is provisioned
* Tests: fix one unittest that fails on Python 3.8+, configure Tox to use Python 3.8, too
* Fix import order
* Don't use 3rd party `mock` package (there is `unittest.mock` and we use it in most places already)
* Improve a test which is unnecessarily strict, breaking when unrelated things change

### CODE REVIEW

**Review by ryanlerch — APPROVED** [review]
LGTM! Just the single conflict to resolve, but other than that. +1

File: devel/create-test-data.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env python3
 import textwrap
 
-import python_freeipa
 from faker import Faker
+import python_freeipa
```
Comment (abompard) [inline devel/create-test-data.py:4]: Are you sure? In [PEP8](https://www.python.org/dev/peps/pep-0008/#id23) they sort imports with `import x` first and `from x import y` second.

In the [Google style guide](https://google.github.io/styleguide/pyguide.html?showone=Imports_formatting#313-imports-formatting) you can see both sortings, and in the [example file](https://github.com/PyCQA/flake8-import-order/blob/master/tests/test_cases/complete_google.py) that flake8-import-order provides the `import x` come before the `from x import y`.

File: tox.ini [hunk 2]
```diff
@@ -74,8 +74,11 @@ commands =
 [flake8]
 show-source = True
 max-line-length = 100
-ignore = E203,E231,W503
 exclude = .git,.tox,dist,*egg
 
+# Configure flake8-import-order
+application-import-names = noggin, noggin_messages
```
Comment (abompard) [inline tox.ini:80]: You need to add `flake8-import-order` to the dev dependencies if you want to use this feature.

**Review by abompard — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 2 above)
Comment (nphilipp) [reply tox.ini:80]: Done.

File: devel/create-test-data.py (hunk 1 above)
Comment (nphilipp) [reply devel/create-test-data.py:4]: I've changed `import-order-style` to `smarkets` in tox.ini which doesn't require 3rd party packages to be separate from each other, but places `from ... import ...` beneath plain `import ...` statements within their respective blocks.

**Review by abompard — APPROVED** [review]

**Pull request merged**